File: 2.php
<?php
// Чтение входных данных
$n = intval(trim(fgets(STDIN)));
$flowersStr = trim(fgets(STDIN));
$capacity = intval(trim(fgets(STDIN)));

// Обработка строки с цветами: заменяем запятые, разбиваем на массив и обрезаем до n элементов
$flowersStr = str_replace(',', ' ', $flowersStr);
$flowers = array_map('intval', preg_split('/\s+/', $flowersStr, -1, PREG_SPLIT_NO_EMPTY));
$flowers = array_slice($flowers, 0, $n); // Фиксируем длину массива

// Инициализация переменных
$steps = 0;
$currentPos = -1; // Начальная позиция у реки
$currentWater = $capacity;

for ($i = 0; $i < $n; $i++) {
    // Шаги до текущего цветка
    $steps += abs($i - $currentPos);
    $currentPos = $i;
    
    // Поливаем цветок
    $currentWater -= $flowers[$i];
    
    // Проверяем необходимость возврата к реке
    if ($i < $n - 1 && $currentWater < $flowers[$i + 1]) {
        $steps += ($currentPos - (-1)); // Шаги до реки
        $currentPos = -1;
        $currentWater = $capacity; // Наполняем лейку
    }
}
echo $steps . PHP_EOL;
?>

File: 2.rs
use std::io;
fn main() {
    // Чтение количества цветов
    let n: usize = read_line().trim().parse().expect("Invalid number");
    // Чтение и обработка строки с потребностями цветов
    let flowers_str = read_line().replace(',', " ");
    let mut flowers: Vec<i32> = flowers_str
        .split_whitespace()
        .map(|s| s.parse().expect("Invalid number"))
        .collect();
    flowers.truncate(n); // Обрезаем до n элементов

    // Чтение вместимости лейки
    let capacity: i32 = read_line().trim().parse().expect("Invalid number");

    let mut steps = 0;
    let mut current_pos = -1; // Начинаем у реки (-1)
    let mut current_water = capacity;

    for i in 0..n {
        // Шаги до текущего цветка
        steps += (i as i32 - current_pos).abs();
        current_pos = i as i32;

        // Поливаем цветок
        current_water -= flowers[i];

        // Проверяем необходимость возврата к реке
        if i < n - 1 && current_water < flowers[i + 1] {
            steps += (current_pos - (-1)).abs(); // Шаги до реки
            current_pos = -1;
            current_water = capacity; // Наполняем лейку
        }
    }

    println!("{}", steps);
}

// Вспомогательная функция для чтения строки из stdin
fn read_line() -> String {
    let mut line = String::new();
    io::stdin().read_line(&mut line).expect("Failed to read line");
    line
}

